### PR TITLE
New elrond client

### PR DESCRIPTION
**Pull request by iulianpascalau**

### CODE REVIEW

File: clients/elrond/transactionHandler.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	proxy                   ElrondProxy
+	relayerAddress          core.AddressHandler
+	multisigAddressAsBech32 string
+	nonceTxHandler          NonceTransactionsHandler
+	relayerPrivateKey       crypto.PrivateKey
+	singleSigner            crypto.SingleSigner
+}
+
+// sendTransactionReturningHash will try to assemble a transaction, sign it, send it and, if everything is OK, returns the transaction's hash
+func (txHandler *transactionHandler) sendTransactionReturningHash(ctx context.Context, builder builders.TxDataBuilder, cost uint64) (string, error) {
```
Comment (sasurobert) [inline clients/elrond/transactionHandler.go:24]: sendTransactionGetHash ? or ReturnHash ?

File: clients/elrond/transactionHandler.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	tx, err := txHandler.signTransaction(ctx, builder, cost)
+	if err != nil {
+		return "", err
+	}
+
+	return txHandler.nonceTxHandler.SendTransaction(context.Background(), tx)
+}
+
+func (txHandler *transactionHandler) signTransaction(ctx context.Context, builder builders.TxDataBuilder, cost uint64) (*data.Transaction, error) {
```
Comment (sasurobert) [inline clients/elrond/transactionHandler.go:33]: rename cost to gasLimit

Comment (sasurobert) [reply clients/elrond/transactionHandler.go:33]: the cost is gasLimit * gasPrice

File: clients/elrond/dataGetter.go [hunk 3]
```diff
@@ -116,17 +116,26 @@ func (dg *dataGetter) ExecuteQueryReturningUint64(ctx context.Context, request *
...
 	}
 
-	num := big.NewInt(0).SetBytes(response[0])
-	if !num.IsUint64() {
+	num, err := parseUInt64FromByteSlice(response[0])
+	if err != nil {
 		return 0, NewQueryResponseError(
 			internalError,
-			"error converting the received bytes to uint64",
+			errNotUint64Bytes.Error(),
```
Comment (sasurobert) [inline clients/elrond/dataGetter.go:123]: why not err.Error ?

File: clients/elrond/transactionHandler.go (hunk 1 above)
Comment (iulianpascalau) [reply clients/elrond/transactionHandler.go:24]: `sendTransactionReturnHash`

File: clients/elrond/transactionHandler.go (hunk 2 above)
Comment (iulianpascalau) [reply clients/elrond/transactionHandler.go:33]: right, copied from the old implementation

File: clients/elrond/dataGetter.go (hunk 3 above)
Comment (iulianpascalau) [reply clients/elrond/dataGetter.go:123]: changed

File: clients/elrond/transactionHandler.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	tx, err := txHandler.signTransaction(ctx, builder, gasLimit)
+	if err != nil {
+		return "", err
+	}
+
+	return txHandler.nonceTxHandler.SendTransaction(context.Background(), tx)
+}
+
+func (txHandler *transactionHandler) signTransaction(ctx context.Context, builder builders.TxDataBuilder, gasLimit uint64) (*data.Transaction, error) {
+	// TODO use a smart cache here (request the configs only from time to time)
```
Comment (dragos-rebegea) [inline clients/elrond/transactionHandler.go:34]: 🔥

**Review by dragos-rebegea — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**